openghg/objectstore/metastore/_metastore.py

```diff
@@ -0,0 +1,242 @@
+"""
+This module defines an interface for metastores.
+
+Metastores store metadata that is used to search an
+object store for data.
+
+The documentation in this module refers to metadata entries stored
+in the metastore as "records". These correspond to "documents" in
+TinyDB and Mongodb.
+
+A MetaStore managed by an ObjectStore object will store
+UUIDs, which are used to locate data stored in the object store.
+
+UUIDs are managed by ObjectStore objects, so they are not part of
+the MetaStore interface.
+
+"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Union
+
+import tinydb
+from tinydb.operations import delete as tinydb_delete
+
+from openghg.types import MetastoreError
+
+MetaData = Dict[str, Any]
```

Comment:
Yea I guess it's not helpful for the user, but it makes the code a bit clearer (in terms of intent). The type aliases could be replaced by actual classes at some point (this would be good for MetaData, not sure about the others).

Reply:
I've left these as is. Besides "MetaData" and "Data", I think most of these will remain internal to the objectstore module, and they could be refined in the future.

For the public API (standardise, etc.) I think the type aliases are a bad idea.